google/resource_google_project.go

```diff
@@ -134,6 +262,33 @@ func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	// set up App Engine, too
+	app, err := expandAppEngineApp(d)
+	if err != nil {
+		return err
+	}
+	if app != nil {
+		log.Printf("[DEBUG] Enabling App Engine")
+		// enable the app engine APIs so we can create stuff
+		if err = enableService("appengine.googleapis.com", project.ProjectId, config); err != nil {
```

Comment:
Won't this mess with people using `google_project_services`?

Reply:
It's a similar approach taken as for disabling the default network, and is documented similarly. I don't know that there's an alternative unless we don't allow people to create App Engine apps in the same apply they create projects :/

Reply:
But wouldn't they already need to enable the resourcemanager api in order to create the project? Couldn't they do that at the same time as the app engine api?

Reply:
APIs get enabled on a per-project basis. The project doesn't exist before the project resource's Create function is called, so we can't enable APIs on it beforehand. But by the time the project's Create function is finished, we need the App Engine Admin API enabled, because we're using it. This is the same rationale, I believe, for enabling the compute API when deleting the default network.

The workaround here, if they want to use `google_project_services`, is to just add the App Engine Admin API to the `google_project_services` resource, which should (I believe) leave things alone if the API is already enabled.

Reply:
Ohhh that makes sense. Thanks!